src/wasm/wasm-emscripten.cpp

```diff
@@ -668,25 +696,40 @@ void AsmConstWalker::visitCall(Call* curr) {
     Builder builder(wasm);
     curr->operands[0] = builder.makeConst(idLiteralForCode(code));
   } else if (import->base.startsWith(INVOKE_PREFIX)) {
+    if (curr->operands.size() < 2) {
+      return;
+    }
     auto idx = resolveConstIndex(curr->operands[0], [&](Expression* arg) {
       Fatal() << "Unexpected table index type (" << getExpressionName(arg)
               << ") in call to: " << import->base;
     });
 
-    // If the address of the invoke call is an emscripten_asm_const_* function:
-    if (asmTable.count(idx)) {
-      auto* value = resolveConstAddr(curr->operands[1], asmTable[idx]);
-      auto code = codeForConstAddr(wasm, segmentOffsets, value);
+    std::string code;
+    Name name;
+    Builder builder(wasm);
+
+    auto fillCodeAndName = [&](const Name& target) {
+      auto* value = resolveConstAddr(curr->operands[1], target);
+      code = codeForConstAddr(wasm, segmentOffsets, value);
 
       // Extract the base signature from the invoke_* function name.
       std::string baseSig(import->base.c_str() + sizeof(INVOKE_PREFIX) - 1);
-      Name name;
       auto sig = fixupNameWithSig(name, baseSig);
       sigsForCode[code].insert(sig);
+    };
 
-      Builder builder(wasm);
+    // If the address of the invoke call is an emscripten_asm_const_* function:
+    if (asmTable.count(idx)) {
+      fillCodeAndName(asmTable[idx]);
       curr->operands[0] = builder.makeConst(tableIndexForName(name));
       curr->operands[1] = builder.makeConst(idLiteralForCode(code));
+    } else {
+      Name* global = resolveGlobalGet(curr->operands[0], import->base);
```

Comment:
It would be good to add a comment explaining what we know if we find ourselves in this case.